src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp

```diff
@@ -527,6 +527,32 @@ class MacroAssembler: public Assembler {
     orr(Vd, T, Vn, Vn);
   }
 
+  // NEON shift instructions
+#define WRAP(INSN)                                                                \
+  void INSN(FloatRegister Vd, SIMD_Arrangement T, FloatRegister Vn, int shift) {  \
+    if (shift == 0) {                                                             \
+        SIMD_Arrangement arrange = (T & 1) == 0 ? T8B : T16B;                     \
+        Assembler::orr(Vd, arrange, Vn, Vn);                                      \
+    } else {                                                                      \
+      Assembler::INSN(Vd, T, Vn, shift);                                          \
+    }                                                                             \
+  }                                                                               \
+
+  WRAP(shl) WRAP(sshr) WRAP(ushr)
+#undef WRAP
+
+#define WRAP(INSN)                                                                \
+  void INSN(FloatRegister Vd, SIMD_Arrangement T, FloatRegister Vn, int shift) {  \
+    if (shift == 0) {                                                             \
+      Assembler::addv(Vd, T, Vd, Vn);                                             \
+    } else {                                                                      \
+      Assembler::INSN(Vd, T, Vn, shift);                                          \
+    }                                                                             \
+  }                                                                               \
+
+  WRAP(usra) WRAP(ssra)
+#undef WRAP
```

Comment:
Are ssra and usra tested by anything? I don't seem them accessed in the test case.

Reply:
Updated. The `ssra/usra` are accessed by tests in `TestVectorShiftImmAndAccumulate.java`.
Manually injected error by changing `addv` to `subv` if shifting right and accumulating with 0, the tests failed as expected.

The `vba.add(vbb.lanewise(SHIFT, Imm))` pattern in `TestVectorShiftImmAndAccumulate.java` are actually the same with the original code in `TestVectorShiftImm.java`.
As of now, I have no idea why `ssra/usra` are not accessed by the previous test code.
The `vba.add(vbb.lanewise(SHIFT, Imm))` pattern should match `ssra/usra` anyway.
I think we need a separate investigation.